crates/core/src/db/durability.rs

```diff
@@ -0,0 +1,178 @@
+use crate::db::persistence::Durability;
+use spacetimedb_commitlog::payload::{
+    txdata::{Mutations, Ops},
+    Txdata,
+};
+use spacetimedb_data_structures::map::IntSet;
+use spacetimedb_datastore::{execution_context::ReducerContext, traits::TxData};
+use spacetimedb_durability::DurableOffset;
+use spacetimedb_primitives::TableId;
+use std::sync::Arc;
+use tokio::sync::mpsc::{unbounded_channel, UnboundedReceiver, UnboundedSender};
+
+/// A request to persist a transaction or to terminate the actor.
+pub enum DurabilityRequest {
+    Work {
+        reducer_context: Option<ReducerContext>,
+        tx_data: Arc<TxData>,
+    },
+    Close,
+}
+
+/// Represents a handle to a background task that persists transactions
+/// according to the [`Durability`] policy provided.
+///
+/// This exists to avoid doing some preparatory work
+/// before sending over to the `Durability` layer.
+#[derive(Clone)]
+pub struct DurabilityWorker {
+    request_tx: UnboundedSender<DurabilityRequest>,
+    durability: Arc<Durability>,
+}
+
+/// Those who run seem to have all the fun... 🎶
+const HUNG_UP: &str = "durability actor hung up / panicked";
+
+impl Drop for DurabilityWorker {
+    fn drop(&mut self) {
+        self.request_tx.send(DurabilityRequest::Close).expect(HUNG_UP);
```

Comment:
I think this needs to just ignore it if the receiving end went away.
There are reasons the real durability worker can go away and then `append_tx` panics.

Not sure though why this would affect the test suite.

Reply:
Apparently we also clone `RelationalDB` in tests which explains the test failure.